enterprise/api_client/sso_orchestrator.py

```diff
@@ -0,0 +1,140 @@
+import base64
+import requests
+
+from http import HTTPStatus
+from urllib.parse import urljoin
+
+from rest_framework.reverse import reverse
+
+from django.conf import settings
+
+from enterprise.errors import ClientError
+from enterprise.utils import (
+    get_configuration_value,
+    get_sso_orchestrator_api_base_url,
+    get_sso_orchestrator_configure_path,
+    get_sso_orchestrator_basic_auth_username,
+    get_sso_orchestrator_basic_auth_password,
+)
+
+class EnterpriseSSOOrchestratorApiClient:
```

Comment:
we may still want to extend from something inside edx-rest-api-client (perhaps a version which supports BasicAuth rather than OAuth) so that we benefit from all the nice default addons like custom user-agents which are app aware, request id forwarding, etc.

Reply:
I didn't see anything about basic auth in the rest api client but I can double check